packages/enzyme/src/ReactWrapper.js

```diff
@@ -357,6 +363,64 @@ class ReactWrapper {
     return this;
   }
 
+  /**
+   * Can be used to update the values provided by React's new context API. A subset of current
+   * providers can be passed to update just those values without removing the unspecified providers.
+   *
+   * NOTE: can only be called on a wrapper instance that is also the root instance.
+   *
+   * Example:
+   * ```
+   * const ContextA = createContext(null);
+   * const ContextB = createContext(null);
+   * const ContextC = createContext(null);
+   *
+   * function MyComponent() {
+   *   return (
+   *     <div>
+   *       <ContextA.Consumer>
+   *         {value => <div>{value} </div>}
+   *       </ContextA.Consumer>
+   *       <ContextB.Consumer>
+   *         {value => <div>{value} </div>}
+   *       </ContextB.Consumer>
+   *       <ContextC.Consumer>
+   *         {value => <div>{value} </div>}
+   *       </ContextC.Consumer>
+   *     </div>
+   *   );
+   * }
+   *
+   * const wrapper = mount(<MyComponent />, {
+   *   providers: [
+   *     <ContextA.Provider value="I" />,
+   *     <ContextB.Provider value="Love" />,
+   *     <ContextC.Provider value="Enzyme" />,
+   *   ],
+   * });
+   * wrapper.text(); // I Love Enzyme
+   *
+   * wrapper.setProviders([
+   *   <ContextA.Provider value="We" />,
+   *   <ContextC.Provider value="React" />,
+   * ]);
+   * wrapper.text(); // We Love React
+   * ```
+   * @param {Array<ReactElement>} providers
+   * @returns {ReactWrapper}
+   */
+  setProviders(providers) {
+    if (this[ROOT] !== this) {
+      throw new Error('ReactWrapper::setProviders() can only be called on the root');
+    }
+    if (!this[RENDERER].setProviders) {
+      throw new Error('ReactWrapper::setProviders() can only be called with react >= 16.3 and a matching enzyme adapter.');
```

Comment:
@ljharb thanks for the clarification on the separation between enzyme itself and its adapters! This is the only file I've changed in the `enzyme` package, so I'm gathering that the problem here is just that I'm mentioning react versions explicitly in this error message (and the one at the top of the file.) Is the solution just to change these errors to only talk about _adapter_ compatibility and not _react_ compatibility?